modules/backend/controllers/Auth.php

```diff
@@ -147,6 +147,19 @@ public function restore()
      */
     public function restore_onSubmit()
     {
+        // Force Trusted Host verification on password reset link generation
+        // regardless of config to protect against host header poisoning
```

Comment:
@LukeTowers I think I would prefer that we don't force it, on the basis that:
a) some people would be opting to configure their web server to protect against this kind of attack and would disable this feature in October CMS to get a small performance increase.
b) it might be a bit misleading to say that `app.trustedHosts` set to `false` disables the feature entirely, just to have it be overridden here regardless.

Reply:
@bennothommo For A we can't depend on the server configuration to solve a security issue this should be implemented in the logic of the software. For B you must do this in order to keep safe from host header poisoning. However, I agree with you about this one it sounds misleading to enable it when the user already disabled it and didn't set Trusted Hosts.

Reply:
@ahussam There are certainly ways of preventing it at the server level - for example, through proxies.

Reply:
@bennothommo I know there is but as an open-source project, this should be fixed inside the project itself rather than depending on external components.

Reply:
@bennothommo personally I'm not super concerned about the performance impact of this, the middleware is loaded regardless, so the actual performance benefit is on the order of 5-20 lines of regular code doing not very much at all, i.e. pretty useless from an optimization standpoint. 

I'm not a huge fan of manually enabling it even when it's explicitly set to false, but it is documented (https://github.com/octobercms/october/pull/5415/files#diff-950397cc5ef29707b54d0c774bd94af85836bab1a27f2f86533d832d25caf224R71-R72). Perhaps we could say that 'null' is the default value and it means that everything is unprotected except for the backend password reset (or even the entire backend), and false is a value that can be used to disable everything.

Reply:
@LukeTowers agreed with the first point.

I'm still favoring a way of disabling it for the second point, but I'm happy to defer and give it a try in this fashion for now. If we get a heap of people getting caught out by it going forward, then I'll push for a change here. Regarding using `null`, I think `null` should probably follow the convention of other configuration options (ie. `false` on dev, `true` on prod).

Reply:
@bennothommo Would it be that useful for this particular config item to have the same null behavior as other items?

Reply:
@LukeTowers it might be useful to have it off for dev, especially for environments where the URL is dynamic (like Octodock).